webapi/Controllers/ChatHistoryController.cs

```diff
@@ -260,4 +268,103 @@ public async Task<IActionResult> GetAllChatSessionsAsync(string userId)
 
         return this.NotFound($"No chat session found for chat id '{chatId}'.");
     }
+
+    /// <summary>
+    /// Delete a chat session.
+    /// </summary>
+    /// <param name="userId">Unique Id of user who initiated the request.</param>
+    /// <param name="sessionId">The chat id.</param>
+    [HttpDelete]
+    [Route("chatSession/{sessionId:guid}")]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [Authorize(Policy = AuthPolicyName.RequireChatParticipant)]
+    public async Task<IActionResult> DeleteChatSessionAsync(
+        [FromServices] IHubContext<MessageRelayHub> messageRelayHubContext,
+        [FromServices] IAuthInfo authInfo,
+        Guid sessionId,
+        CancellationToken cancellationToken)
+    {
+        var chatId = sessionId.ToString();
+        ChatSession? chatToDelete = null;
+        try
+        {
+            // Make sure the chat session exists
+            chatToDelete = await this._sessionRepository.FindByIdAsync(chatId);
+        }
+        catch (KeyNotFoundException)
+        {
+            return this.NotFound($"No chat session found for chat id '{chatId}'.");
+        }
+
+        // Delete any resources associated with the chat session.
+        var deleteResourcesResult = await this.DeleteChatResourcesAsync(sessionId, cancellationToken) as StatusCodeResult;
+        if (deleteResourcesResult?.StatusCode != 204)
+        {
+            return this.StatusCode(500, $"Failed to delete resources for chat id '{chatId}'.");
+        }
+
+        // Delete chat session and broadcast update to all participants.
+        await this._sessionRepository.DeleteAsync(chatToDelete);
+        await messageRelayHubContext.Clients.Group(chatId).SendAsync(ChatDeletedClientCall, chatId, authInfo.UserId, cancellationToken: cancellationToken);
+
+        return this.NoContent();
+    }
+
+    /// <summary>
+    /// Deletes all associated resources (messages, memories, participants) associated with a chat session.
+    /// </summary>
+    /// <param name="sessionId">The chat id.</param>
+    private async Task<IActionResult> DeleteChatResourcesAsync(Guid sessionId, CancellationToken cancellationToken)
+    {
+        var chatId = sessionId.ToString();
+        var cleanupTasks = new List<Task>();
+
+        // Create and store the tasks for deleting all users tied to the chat.
+        var participants = await this._participantRepository.FindByChatIdAsync(chatId);
+        foreach (var participant in participants)
+        {
+            cleanupTasks.Add(this._participantRepository.DeleteAsync(participant));
```

Comment:
Just confirming...  These are the opaque GUID users, right?

Reply:
What do you mean by opaque?

Reply:
It's the default user Id, if that's what you mean. There should only be one chat participant per chat session if the user is in Guest mode

![image](https://github.com/microsoft/chat-copilot/assets/125500434/f95e69a4-0442-49bb-b0d2-8f6db1067e2b)

![image](https://github.com/microsoft/chat-copilot/assets/125500434/c92a0748-b2fd-4756-a7f8-fce14d7c37e3)

Reply:
OK. I'm good with that!